### PR TITLE
Document the change to a spinlock in the change log

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

**Comment by JackStouffer** [comment]
ping @MartinNowak

File: changelog.dd [hunk 1]
```diff
@@ -9,6 +9,10 @@ $(LI $(RELATIVE_LINK2 aa-clear, A `clear` method has been added to associative
     arrays to remove all elements.))
 )
 
+$(LI $(RELATIVE_LINK2 spinlock,The GC is no longer wrapped in a pthread_mutex,
```
Comment (MartinNowak) [inline changelog.dd]: The GC now uses a spinlock instead of a recursive mutex.

File: changelog.dd [hunk 2]
```diff
@@ -27,6 +31,15 @@ aa.clear(); // still retains its current capacity for faster imports
 assert(aa2.length == 0); // other references affected
 -------
 )
+
+$(LI $(LNAME2 spinlock, The GC is no longer wrapped in a pthread_mutex, it
+    now uses a spinlock.)
+
+    $(P This results in a 5% faster GC on average, with the most benefits going
+        to multi-threaded programs that use the GC. See
+        $(WEB https://github.com/D-Programming-Language/druntime/pull/1447#issue-119621375,
```
Comment (MartinNowak) [inline changelog.dd]: Better add a DRUNTIMEPR macro below, see https://github.com/D-Programming-Language/dmd/blob/7cf24a4b5f07f920c0d772ac84a3c5547db2c538/changelog.dd#L138.

**Comment by JackStouffer** [comment]
Fixed

**Comment by JackStouffer** [comment]
ping @MartinNowak

**Pull request merged**